src/include/txn_inline.h

```diff
@@ -695,9 +695,9 @@ __wt_txn_op_set_timestamp(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool validate
             __wt_txn_op_delete_apply_prepare_state(session, op, true);
         else {
             upd = op->u.op_upd;
-
-            /* Resolve prepared update to be committed update. */
-            __txn_apply_prepare_state_update(session, upd, true);
+            if (upd != NULL)
```

Comment:
can you add a comment here in explaining in what scenarios the update can be NULL.